include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2679,8 +2680,15 @@ namespace internal
       {
         if (integrate)
           proc.function_0(temp1, comp);
+
+        // we can only use the fast functions if we know the polynomial degree
+        // as a template parameter (fe_degree != -1), and it only makes sense
+        // to use the functions for at least linear functions for values on
+        // the faces and quadratic functions for gradients on the faces, so
+        // include the switch here
         if ((do_gradients == false &&
-             data.data.front().nodal_at_cell_boundaries == true) ||
+             data.data.front().nodal_at_cell_boundaries == true &&
+             fe_degree > 0) ||
```

Comment:
This is the essential addition - the `do_gradients == false && data.data.front().nodal_at_cell_boundaries == true)` would run here with `fe_degree = -1` which would not work with some compile-time offsets below. The rest of the changes are just the static loop lengths.